Improve consistency with openapi-generator

jhannes opened the pull request:
I have four commits:

* allowableValues.enumvars was set on model and not on property, meaning that a model with two enums would fial
* property.isString and property.itemsComplexType = 'string' will improve compatibility with openapi-generator templates
* I wanted to use Handlebars for some templates. By adding render to the generator specified in the configFile, you can plug in any templating system you want
* operation.requestBody didn't support application/x-www-form-urlencoded

I tried to make the commits as small as possible, but I realize that the use of confiig.generator.render probably would be better by making Hogan into a default and using config.generator.render instead of Hogan in adaptor.transform

MikeRalphson commented:
Thanks, looking pretty good. Unsure why it's not linked to this PR, but could you take a look at the Travis test failures: https://travis-ci.org/github/Mermade/openapi-codegen/jobs/705088675

MikeRalphson commented:
Still failing CI https://travis-ci.org/github/Mermade/openapi-codegen/jobs/709565665

jhannes commented:
Sorry about the late response. I was on vacation. 🌞

Thanks for pointing out the build problem for me. It turns out I was a bit sloppy with undef checking. I've fixed this and the tests now seem to pass.

I also added a few notes about testing to CONTRIBUTING.md

jhannes commented:
Seems like there is a problem with the generation to Lua and Go, two of the last environments I would consider checking

MikeRalphson commented on CONTRIBUTING.md:
```diff
@@ -6,6 +6,21 @@ First of all, many thanks for considering making a contribution to this project.
+
+1. `npm run lint`
+1. `npm test`
+1. `bin/buildAll.sh`
```
I would prefer **not** to have this information in the CONTRIBUTING.md apart from the first two - (ignoring current Travis problems), as it should not be necessary for people to run them locally as a first resort, that's what the CI run is designed to catch, and it would have to be kept in sync as more language tests were added.

jhannes replied:
Done

MikeRalphson commented:
> Seems like there is a problem with the generation to Lua and Go, two of the last environments I would consider checking

Not sure I understand that? These tests were not failing prior to this PR though.

jhannes commented:
I found the problems - there are a lot of assumptions that I don't yet understand when it comes to the internal structure and I had forgotten to update the variables `operation.hasRequiredParams` and `operation.hasOptionalParams` when processing form variables. I also didn't clone the parameters before putting it in allParams and formParams, so the "-last" and "-first" flags weren't handled correctly.

Travis is happy now :-)

MikeRalphson commented:
Many thanks for your work on this. Will have another look over tomorrow and merge. Thanks again!

MikeRalphson commented:
Thanks again!
